<?xml version="1.0" encoding="UTF-8"?>
<testsuites>
  <testsuite name="" tests="31" assertions="161" errors="0" warnings="0" failures="2" skipped="0" time="7.871335">
    <testsuite name="Project Test Suite" tests="31" assertions="161" errors="0" warnings="0" failures="2" skipped="0" time="7.871335">
      <testsuite name="App\Tests\Functional\Controller\AuthorizationControllerTest" file="/app/tests/Functional/Controller/AuthorizationControllerTest.php" tests="5" assertions="16" errors="0" warnings="0" failures="0" skipped="0" time="4.644984">
        <testsuite name="App\Tests\Functional\Controller\AuthorizationControllerTest::testLogin" tests="2" assertions="6" errors="0" warnings="0" failures="0" skipped="0" time="2.659045">
          <testcase name="testLogin with data set &quot;success&quot;" class="App\Tests\Functional\Controller\AuthorizationControllerTest" classname="App.Tests.Functional.Controller.AuthorizationControllerTest" file="/app/tests/Functional/Controller/AuthorizationControllerTest.php" line="52" assertions="3" time="2.039139"/>
          <testcase name="testLogin with data set &quot;missing data&quot;" class="App\Tests\Functional\Controller\AuthorizationControllerTest" classname="App.Tests.Functional.Controller.AuthorizationControllerTest" file="/app/tests/Functional/Controller/AuthorizationControllerTest.php" line="52" assertions="3" time="0.619906"/>
        </testsuite>
        <testcase name="testLogoutSuccess" class="App\Tests\Functional\Controller\AuthorizationControllerTest" classname="App.Tests.Functional.Controller.AuthorizationControllerTest" file="/app/tests/Functional/Controller/AuthorizationControllerTest.php" line="82" assertions="3" time="0.771026"/>
        <testsuite name="App\Tests\Functional\Controller\AuthorizationControllerTest::testRegisterSuccess" tests="2" assertions="7" errors="0" warnings="0" failures="0" skipped="0" time="1.214913">
          <testcase name="testRegisterSuccess with data set &quot;success&quot;" class="App\Tests\Functional\Controller\AuthorizationControllerTest" classname="App.Tests.Functional.Controller.AuthorizationControllerTest" file="/app/tests/Functional/Controller/AuthorizationControllerTest.php" line="117" assertions="4" time="0.653030"/>
          <testcase name="testRegisterSuccess with data set &quot;missing data&quot;" class="App\Tests\Functional\Controller\AuthorizationControllerTest" classname="App.Tests.Functional.Controller.AuthorizationControllerTest" file="/app/tests/Functional/Controller/AuthorizationControllerTest.php" line="117" assertions="3" time="0.561883"/>
        </testsuite>
      </testsuite>
      <testsuite name="App\Tests\Functional\Controller\PurchaseControllerTest" file="/app/tests/Functional/Controller/PurchaseControllerTest.php" tests="2" assertions="11" errors="0" warnings="0" failures="1" skipped="0" time="1.365492">
        <testcase name="testPurchase" class="App\Tests\Functional\Controller\PurchaseControllerTest" classname="App.Tests.Functional.Controller.PurchaseControllerTest" file="/app/tests/Functional/Controller/PurchaseControllerTest.php" line="37" assertions="1" time="0.664962">
          <failure type="PHPUnit\Framework\ExpectationFailedException">App\Tests\Functional\Controller\PurchaseControllerTest::testPurchase
Failed asserting that 403 matches expected 200.

/app/tests/Functional/Controller/PurchaseControllerTest.php:72
/app/vendor/phpunit/phpunit/phpunit:107</failure>
        </testcase>
        <testcase name="testGetPurchasedGamesSuccess" class="App\Tests\Functional\Controller\PurchaseControllerTest" classname="App.Tests.Functional.Controller.PurchaseControllerTest" file="/app/tests/Functional/Controller/PurchaseControllerTest.php" line="102" assertions="10" time="0.700530"/>
      </testsuite>
      <testsuite name="App\Tests\Functional\Controller\UserInfoControllerTest" file="/app/tests/Functional/Controller/UserInfoControllerTest.php" tests="1" assertions="15" errors="0" warnings="0" failures="0" skipped="0" time="0.613975">
        <testcase name="testGetUserInfoSuccess" class="App\Tests\Functional\Controller\UserInfoControllerTest" classname="App.Tests.Functional.Controller.UserInfoControllerTest" file="/app/tests/Functional/Controller/UserInfoControllerTest.php" line="39" assertions="15" time="0.613975"/>
      </testsuite>
      <testsuite name="App\Tests\Functional\Repository\UserRepositoryTest" file="/app/tests/Functional/Repository/UserRepositoryTest.php" tests="2" assertions="2" errors="0" warnings="0" failures="0" skipped="0" time="1.052804">
        <testcase name="testUpgradePasswordSuccess" class="App\Tests\Functional\Repository\UserRepositoryTest" classname="App.Tests.Functional.Repository.UserRepositoryTest" file="/app/tests/Functional/Repository/UserRepositoryTest.php" line="31" assertions="1" time="0.532868"/>
        <testcase name="testUpgradePasswordNotSupported" class="App\Tests\Functional\Repository\UserRepositoryTest" classname="App.Tests.Functional.Repository.UserRepositoryTest" file="/app/tests/Functional/Repository/UserRepositoryTest.php" line="51" assertions="1" time="0.519937"/>
      </testsuite>
      <testsuite name="App\Tests\Unit\Entity\GameTest" file="/app/tests/Unit/Entity/GameTest.php" tests="2" assertions="10" errors="0" warnings="0" failures="0" skipped="0" time="0.002535">
        <testcase name="testCreateEmptyGameEntity" class="App\Tests\Unit\Entity\GameTest" classname="App.Tests.Unit.Entity.GameTest" file="/app/tests/Unit/Entity/GameTest.php" line="11" assertions="4" time="0.001108"/>
        <testcase name="testCreateNotEmptyGameEntity" class="App\Tests\Unit\Entity\GameTest" classname="App.Tests.Unit.Entity.GameTest" file="/app/tests/Unit/Entity/GameTest.php" line="21" assertions="6" time="0.001427"/>
      </testsuite>
      <testsuite name="App\Tests\Unit\Entity\PurchasedGameTest" file="/app/tests/Unit/Entity/PurchasedGameTest.php" tests="2" assertions="7" errors="0" warnings="0" failures="0" skipped="0" time="0.002391">
        <testcase name="testCreateEmptyPurchasedGame" class="App\Tests\Unit\Entity\PurchasedGameTest" classname="App.Tests.Unit.Entity.PurchasedGameTest" file="/app/tests/Unit/Entity/PurchasedGameTest.php" line="12" assertions="4" time="0.001073"/>
        <testcase name="testCreateNotEmptyPurchasedGame" class="App\Tests\Unit\Entity\PurchasedGameTest" classname="App.Tests.Unit.Entity.PurchasedGameTest" file="/app/tests/Unit/Entity/PurchasedGameTest.php" line="22" assertions="3" time="0.001317"/>
      </testsuite>
      <testsuite name="App\Tests\Unit\Entity\UserTest" file="/app/tests/Unit/Entity/UserTest.php" tests="2" assertions="18" errors="0" warnings="0" failures="0" skipped="0" time="0.006726">
        <testcase name="testCreateEmptyUserEntity" class="App\Tests\Unit\Entity\UserTest" classname="App.Tests.Unit.Entity.UserTest" file="/app/tests/Unit/Entity/UserTest.php" line="11" assertions="6" time="0.001616"/>
        <testcase name="testCreateNotEmptyUserEntity" class="App\Tests\Unit\Entity\UserTest" classname="App.Tests.Unit.Entity.UserTest" file="/app/tests/Unit/Entity/UserTest.php" line="22" assertions="12" time="0.005110"/>
      </testsuite>
      <testsuite name="App\Tests\Unit\Security\ApiTokenFailureHandlerTest" file="/app/tests/Unit/Security/ApiTokenFailureHandlerTest.php" tests="1" assertions="1" errors="0" warnings="0" failures="0" skipped="0" time="0.007810">
        <testcase name="testOnAuthenticationFailure" class="App\Tests\Unit\Security\ApiTokenFailureHandlerTest" classname="App.Tests.Unit.Security.ApiTokenFailureHandlerTest" file="/app/tests/Unit/Security/ApiTokenFailureHandlerTest.php" line="14" assertions="1" time="0.007810"/>
      </testsuite>
      <testsuite name="App\Tests\Unit\Security\ApiTokenHandlerTest" file="/app/tests/Unit/Security/ApiTokenHandlerTest.php" tests="3" assertions="11" errors="0" warnings="0" failures="0" skipped="0" time="0.092903">
        <testsuite name="App\Tests\Unit\Security\ApiTokenHandlerTest::testGetUserBadgeFrom" tests="3" assertions="11" errors="0" warnings="0" failures="0" skipped="0" time="0.092903">
          <testcase name="testGetUserBadgeFrom with data set &quot;success&quot;" class="App\Tests\Unit\Security\ApiTokenHandlerTest" classname="App.Tests.Unit.Security.ApiTokenHandlerTest" file="/app/tests/Unit/Security/ApiTokenHandlerTest.php" line="41" assertions="4" time="0.081056"/>
          <testcase name="testGetUserBadgeFrom with data set &quot;bad credentials&quot;" class="App\Tests\Unit\Security\ApiTokenHandlerTest" classname="App.Tests.Unit.Security.ApiTokenHandlerTest" file="/app/tests/Unit/Security/ApiTokenHandlerTest.php" line="41" assertions="3" time="0.005690"/>
          <testcase name="testGetUserBadgeFrom with data set &quot;expired token&quot;" class="App\Tests\Unit\Security\ApiTokenHandlerTest" classname="App.Tests.Unit.Security.ApiTokenHandlerTest" file="/app/tests/Unit/Security/ApiTokenHandlerTest.php" line="41" assertions="4" time="0.006157"/>
        </testsuite>
      </testsuite>
      <testsuite name="App\Tests\Unit\Service\AuthorizationServiceTest" file="/app/tests/Unit/Service/AuthorizationServiceTest.php" tests="7" assertions="31" errors="0" warnings="0" failures="0" skipped="0" time="0.043335">
        <testsuite name="App\Tests\Unit\Service\AuthorizationServiceTest::testLogin1" tests="3" assertions="13" errors="0" warnings="0" failures="0" skipped="0" time="0.023564">
          <testcase name="testLogin1 with data set &quot;success&quot;" class="App\Tests\Unit\Service\AuthorizationServiceTest" classname="App.Tests.Unit.Service.AuthorizationServiceTest" file="/app/tests/Unit/Service/AuthorizationServiceTest.php" line="70" assertions="5" time="0.010563"/>
          <testcase name="testLogin1 with data set &quot;user does not exist&quot;" class="App\Tests\Unit\Service\AuthorizationServiceTest" classname="App.Tests.Unit.Service.AuthorizationServiceTest" file="/app/tests/Unit/Service/AuthorizationServiceTest.php" line="70" assertions="4" time="0.006529"/>
          <testcase name="testLogin1 with data set &quot;wrong password&quot;" class="App\Tests\Unit\Service\AuthorizationServiceTest" classname="App.Tests.Unit.Service.AuthorizationServiceTest" file="/app/tests/Unit/Service/AuthorizationServiceTest.php" line="70" assertions="4" time="0.006472"/>
        </testsuite>
        <testcase name="testRegisterSuccess" class="App\Tests\Unit\Service\AuthorizationServiceTest" classname="App.Tests.Unit.Service.AuthorizationServiceTest" file="/app/tests/Unit/Service/AuthorizationServiceTest.php" line="114" assertions="3" time="0.002853"/>
        <testsuite name="App\Tests\Unit\Service\AuthorizationServiceTest::testLogout" tests="3" assertions="15" errors="0" warnings="0" failures="0" skipped="0" time="0.016918">
          <testcase name="testLogout with data set &quot;success&quot;" class="App\Tests\Unit\Service\AuthorizationServiceTest" classname="App.Tests.Unit.Service.AuthorizationServiceTest" file="/app/tests/Unit/Service/AuthorizationServiceTest.php" line="130" assertions="5" time="0.005786"/>
          <testcase name="testLogout with data set &quot;user does not exist&quot;" class="App\Tests\Unit\Service\AuthorizationServiceTest" classname="App.Tests.Unit.Service.AuthorizationServiceTest" file="/app/tests/Unit/Service/AuthorizationServiceTest.php" line="130" assertions="5" time="0.005517"/>
          <testcase name="testLogout with data set &quot;user already unauthorized&quot;" class="App\Tests\Unit\Service\AuthorizationServiceTest" classname="App.Tests.Unit.Service.AuthorizationServiceTest" file="/app/tests/Unit/Service/AuthorizationServiceTest.php" line="130" assertions="5" time="0.005614"/>
        </testsuite>
      </testsuite>
      <testsuite name="App\Tests\Unit\Service\PurchaseServiceTest" file="/app/tests/Unit/Service/PurchaseServiceTest.php" tests="3" assertions="22" errors="0" warnings="0" failures="1" skipped="0" time="0.033182">
        <testsuite name="App\Tests\Unit\Service\PurchaseServiceTest::testPurchase" tests="2" assertions="10" errors="0" warnings="0" failures="0" skipped="0" time="0.027159">
          <testcase name="testPurchase with data set &quot;success&quot;" class="App\Tests\Unit\Service\PurchaseServiceTest" classname="App.Tests.Unit.Service.PurchaseServiceTest" file="/app/tests/Unit/Service/PurchaseServiceTest.php" line="53" assertions="5" time="0.020976"/>
          <testcase name="testPurchase with data set &quot;game already purchased&quot;" class="App\Tests\Unit\Service\PurchaseServiceTest" classname="App.Tests.Unit.Service.PurchaseServiceTest" file="/app/tests/Unit/Service/PurchaseServiceTest.php" line="53" assertions="5" time="0.006183"/>
        </testsuite>
        <testcase name="testGetPurchasedGamesSuccess" class="App\Tests\Unit\Service\PurchaseServiceTest" classname="App.Tests.Unit.Service.PurchaseServiceTest" file="/app/tests/Unit/Service/PurchaseServiceTest.php" line="93" assertions="12" time="0.006024">
          <failure type="PHPUnit\Framework\ExpectationFailedException">App\Tests\Unit\Service\PurchaseServiceTest::testGetPurchasedGamesSuccess
Expectation failed for method name is "decodeLongToken" when invoked 1 time(s).
Method was expected to be called 1 times, actually called 0 times.

/app/vendor/phpunit/phpunit/phpunit:107</failure>
        </testcase>
      </testsuite>
      <testsuite name="App\Tests\Unit\Service\UserInfoServiceTest" file="/app/tests/Unit/Service/UserInfoServiceTest.php" tests="1" assertions="17" errors="0" warnings="0" failures="0" skipped="0" time="0.005197">
        <testcase name="testGetUserInfo" class="App\Tests\Unit\Service\UserInfoServiceTest" classname="App.Tests.Unit.Service.UserInfoServiceTest" file="/app/tests/Unit/Service/UserInfoServiceTest.php" line="22" assertions="17" time="0.005197"/>
      </testsuite>
    </testsuite>
  </testsuite>
</testsuites>
